ecommerce/programs/api.py

```
@@ -0,0 +1,35 @@
+from django.conf import settings
+from django.core.cache import cache
+
+
+class ProgramsApiClient(object):
+    """ Client for the Programs API.
+
+    This client is designed to cache the data retrieved from the Programs API to
+    reduce load on the API and increase performance of consuming services.
+    """
+
+    def __init__(self, client):
+        self.cache_ttl = settings.PROGRAM_CACHE_TIMEOUT
+        self.client = client
+
+    def get_program(self, uuid):
+        """
+        Retrieve the details for a single program.
+
+        Args:
+            uuid (str|uuid): Program UUID.
+
+        Returns:
+            dict
+        """
+        program_uuid = str(uuid)
+        cache_key = 'program-{uuid}'.format(uuid=program_uuid)
+
+        program = cache.get(cache_key)
+
+        if not program:
+            program = self.client.programs(program_uuid).get()
```

We concluded that it's unnecessary. Instead we will let it bubble up since it's not expected to happen often but if it happens the edx staff knows how to report issues.